### ISSUE TITLE
setUp template proposed by calypso could contain super call

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#6329]
When we browse a test class calypso proposes a pane with setUp and a nice comment. 
It would be nice that the template also includes

```
setUp
     super setUp.
```

**Comment by VincentBlondeau** [comment]
And the same for the tearDown method:
```
tearDown 
     super tearDown.
```

**Comment by dionisiydk** [comment]
Yes, that would be nice.

But notice that currently what you see is not a template but the actual
method TestCase>>#setUp (or whatever #setUp exists in current test
hierarchy). So special logic needs to be added to override "default setup"
method with special "template sources".

вс, 10 мая 2020 г. в 19:55, StéphaneDucasse <notifications@github.com>:

**Comment by dionisiydk** [comment]
вс, 10 мая 2020 г. в 20:17, Vincent Blondeau <notifications@github.com>:

> And the same for the tearDown method:
>
> tearDown
>      super tearDown.
>
> There is no behaviour in Calypso to assist with #tearDown method. Only
#setUp tab is available

**Comment by Ducasse** [comment]
Yes denis I was talking about this one.

**Comment by olekscode** [comment]
Let me clarify this a bit.

At the moment, Calypso does not provide any template for setUp and tearDown methods.
"setUp" and "tearDown" tabs appear only for those classes that have those methods implemented.
You want to show those tabs for every class. And in case the class does not have it's own implementation of setUp or tearDown, the tab will show the default template:
```Smalltalk
setUp
    super setUp.
```
```Smalltalk
tearDown
    super tearDown.
```
Am I correct?

**Comment by olekscode** [comment]
We changed the implementation of `ClyTestSetUpEditorTool` in such a way that setUp tab is only displayed for the test classes that provide their own implementation of setUp.
Before, there was a lookup and the implementation of setUp from the nearest superclass was displayed (which in most cases was the  empty method with a comment from `TestCase`). Now we only check the child class which is being displayed and don't do the lookup.

The tearDown tab is not displayed.
However, I believe that it should be treated in the same way as setUp.
So I have created a separate issue for it: #6705
